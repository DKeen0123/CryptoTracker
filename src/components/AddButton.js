import React from 'react';

const AddButton = props => {
  return <button onClick={props.addCrypto}>{props.label}</button>;
};

export default AddButton;