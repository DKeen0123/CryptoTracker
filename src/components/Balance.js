import React from 'react';

const Balance = () => {
  return (
    <div>
      <h4>Your balance: </h4>
    </div>
  );
};

export default Balance;
