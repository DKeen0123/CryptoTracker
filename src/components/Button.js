import React from 'react';

const Button = () => {
  return <button>Deposit</button>;
};

export default Button;
