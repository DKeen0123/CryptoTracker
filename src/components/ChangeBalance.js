import React from 'react';

const ChangeBalance = () => {
  return (
    <div>
      <input />
    </div>
  );
};

export default ChangeBalance;
