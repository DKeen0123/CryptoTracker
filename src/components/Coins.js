import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/action_bitcoin';
import { fetchEthereum } from '../actions/action_ethereum';

export class Coins extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
    this.props.fetchEthereum();
  }

  calculateBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return '';
    return this.props.balance / parseInt(bitcoin[0].price_usd, 10);
  }

  calculateEthereum() {
    const { ethereum } = this.props;
    if (ethereum === null || ethereum === undefined) return '';
    if (Object.keys(ethereum).length === 0) return '';
    return this.props.balance / parseInt(ethereum[0].price_usd, 10);
  }

  render() {
    return (
      <div>
        <h3 className="bitcoin-balance">Bitcoin: {this.calculateBitcoin()}</h3>
        <h3 className="ethereum-balance">
          Ethereum: {this.calculateEthereum()}
        </h3>
      </div>
    );
  }
}

export default connect(state => state, { fetchBitcoin, fetchEthereum })(Coins);
