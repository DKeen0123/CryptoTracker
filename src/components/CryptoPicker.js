import React, { Component } from 'react';

class CryptoPicker extends Component {
  constructor(props) {
    super()
  }
  populateCryptos() {
    const { cryptos } = this.props;
    if (cryptos === null || cryptos === undefined) return '';
    if (Object.keys(cryptos).length === 0) return '';
    return cryptos.map(crypto => {
      return (
        <option id={crypto.id} key={crypto.id}>
          {crypto.name}
        </option>
      );
    });
  }

  render() {
    return (
      <div>
        <h3>Choose your crypto: </h3>
        <select
          onChange={this.props.selectCrypto}
          className="drop-down"
          value={this.props.chosenCrypto}
        >
          {this.populateCryptos()}
        </select>

      </div>
    );
  }
}

export default CryptoPicker;