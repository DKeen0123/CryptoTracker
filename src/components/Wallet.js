import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
  constructor() {
    super();

    this.state = { balance: undefined };
  }

  updateBalance = event =>
    this.setState({ balance: parseInt(event.target.value, 10) });

  render() {
    return (
      <div>
        <h3 className="balance">Balance: {this.props.balance}</h3>
        <input className="input-balance" onChange={this.updateBalance} />
      </div>
    );
  }
}

export default connect(state => {
  return { balance: state.balance };
}, null)(Wallet);
