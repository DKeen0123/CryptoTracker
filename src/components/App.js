import React from 'react';
import Wallet from './Wallet';
import Coins from './Coins';
import CryptoPicker from './CryptoPicker';

const App = () => {
  return (
    <div>
      <h2>CryptoTracker</h2>
      <hr />
      <Wallet />
      <CryptoPicker />
      <Coins />
    </div>
  );
};

export default App;
