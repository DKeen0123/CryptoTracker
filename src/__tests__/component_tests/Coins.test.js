import React from 'react';
import { shallow, mount } from 'enzyme';
import { Coins } from '../../components/Coins';

describe('Coins', () => {
  let props = { balance: 10, bitcoin: {} };
  let coins = shallow(<Coins {...props} />);

  it('renders correctly', () => {
    expect(coins).toMatchSnapshot();
  });

  describe('when mounted', () => {
    const mockFetchBitcoin = jest.fn();
    const mockFetchEthereum = jest.fn();
    beforeEach(() => {
      props.fetchBitcoin = mockFetchBitcoin;
      props.fetchEthereum = mockFetchEthereum;
      coins = mount(<Coins {...props} />);
    });

    it('dispatches the `fetchBitcoin()` method received from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });

    it('dispatches the `fetchEthereum()` method received from props', () => {
      expect(mockFetchEthereum).toHaveBeenCalled();
    });

    describe('when there are valid bitcoin props', () => {
      beforeEach(() => {
        props = { balance: 10, bitcoin: [{ price_usd: '1000' }] };
        coins = shallow(<Coins {...props} />);
      });

      it('renders the correct bitcoin value to the screen', () => {
        expect(coins.find('.bitcoin-balance').text()).toEqual('Bitcoin: 0.01');
      });
    });

    describe('when there are valid ethereum props', () => {
      beforeEach(() => {
        props = {
          balance: 5,
          bitcoin: [{ price_usd: '1000' }],
          ethereum: [{ price_usd: '500' }]
        };
        coins = shallow(<Coins {...props} />);
      });

      it('renders the correct ethereum value to the screen', () => {
        expect(coins.find('.ethereum-balance').text()).toEqual(
          'Ethereum: 0.01'
        );
      });
    });
  });
});
