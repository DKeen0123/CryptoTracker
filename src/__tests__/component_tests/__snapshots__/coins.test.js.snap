// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Coins
    balance={10}
    bitcoin={Object {}}
    selectedCryptos={
        Array [
            "XRP",
            "TRON",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="bitcoin-balance"
>
          Bitcoin: 
          
</h3>,
        <h3
          className="ethereum-balance"
>
          Ethereum: 
          
</h3>,
        Array [
          <h3
            id="XRP"
>
            XRP
            :
</h3>,
          <h3
            id="TRON"
>
            TRON
            :
</h3>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bitcoin: ",
            "",
          ],
          "className": "bitcoin-balance",
        },
        "ref": null,
        "rendered": Array [
          "Bitcoin: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ethereum: ",
            "",
          ],
          "className": "ethereum-balance",
        },
        "ref": null,
        "rendered": Array [
          "Ethereum: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "XRP",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "XRP",
            ":",
          ],
          "id": "XRP",
        },
        "ref": null,
        "rendered": Array [
          "XRP",
          ":",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "TRON",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "TRON",
            ":",
          ],
          "id": "TRON",
        },
        "ref": null,
        "rendered": Array [
          "TRON",
          ":",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="bitcoin-balance"
>
            Bitcoin: 
            
</h3>,
          <h3
            className="ethereum-balance"
>
            Ethereum: 
            
</h3>,
          Array [
            <h3
              id="XRP"
>
              XRP
              :
</h3>,
            <h3
              id="TRON"
>
              TRON
              :
</h3>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bitcoin: ",
              "",
            ],
            "className": "bitcoin-balance",
          },
          "ref": null,
          "rendered": Array [
            "Bitcoin: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ethereum: ",
              "",
            ],
            "className": "ethereum-balance",
          },
          "ref": null,
          "rendered": Array [
            "Ethereum: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "XRP",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "XRP",
              ":",
            ],
            "id": "XRP",
          },
          "ref": null,
          "rendered": Array [
            "XRP",
            ":",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "TRON",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "TRON",
              ":",
            ],
            "id": "TRON",
          },
          "ref": null,
          "rendered": Array [
            "TRON",
            ":",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
